core/src/components/tab-button/tab-button.ios.scss

```
@@ -26,6 +26,64 @@
   line-height: 16px;
 }
 
+:host ::slotted(ion-badge:empty) {
+  @include padding(0, 0);
+
+  width: $tab-ios-button-badge-size-emtpy;
+  height: $tab-ios-button-badge-size-emtpy;
+}
+
+:host ::slotted(ion-badge[vertical]) {
+  @include position(null, null, null, $tab-ios-button-badge-start-position);
+}
+
+:host ::slotted(ion-badge.badge-vertical-top) {
+  @include position($tab-ios-button-badge-top-position, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-top:empty) {
+  @include position($tab-ios-button-badge-top-position-empty, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-bottom) {
+  @include position($tab-ios-button-badge-bottom-position, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-bottom:empty) {
+  @include position($tab-ios-button-badge-bottom-position-empty, null, null, null);
+}
```

Since we don't change any of the other value, we can depend on the defaults. I recommend reviewing how `position` works.

```suggestion
:host ::slotted(ion-badge.badge-vertical-top) {
  @include position($tab-ios-button-badge-top-position);
}

:host ::slotted(ion-badge.badge-vertical-top:empty) {
  @include position($tab-ios-button-badge-top-position-empty);
}

:host ::slotted(ion-badge.badge-vertical-bottom) {
  @include position($tab-ios-button-badge-bottom-position);
}

:host ::slotted(ion-badge.badge-vertical-bottom:empty) {
  @include position($tab-ios-button-badge-bottom-position-empty);
}
```